src/circle_sel.c

```diff
@@ -0,0 +1,218 @@
+#include "circle.h"
+#include <utils/selfuncs.h>
+#include <utils/lsyscache.h>
+
+/* Circle selectivity functions */
+
+PG_FUNCTION_INFO_V1(spherepoint_in_circle_sel);
+PG_FUNCTION_INFO_V1(spherepoint_in_circle_joinsel);
+PG_FUNCTION_INFO_V1(spherepoint_in_circle_neg_sel);
+PG_FUNCTION_INFO_V1(spherepoint_in_circle_neg_joinsel);
+PG_FUNCTION_INFO_V1(spherepoint_in_circle_com_sel);
+PG_FUNCTION_INFO_V1(spherepoint_in_circle_com_joinsel);
+PG_FUNCTION_INFO_V1(spherepoint_in_circle_com_neg_sel);
+PG_FUNCTION_INFO_V1(spherepoint_in_circle_com_neg_joinsel);
+
+/*
+ * Common code for spherepoint_in_circle_sel()
+ */
+static double
+spherepoint_in_circle_sel_funcexpr(Node *other)
+{
+	FuncExpr *ofunc = (FuncExpr *)other;
+	char *func_name = get_func_name(ofunc->funcid);
+	Const *arg2;
+	double radius;
```

Comment:
One line what?

Reply:
Ah I think I understand what you mean, omit the `ofunc` definition and inline it on the next line. That won't work since ofunc is used several times later on. Plus it's a cast expression that is really better readable when the type is made explicit.